### ISSUE TITLE
Issue causing map to not load.

### ISSUE THREAD

**Issue by rolinger** [dpa99c/cordova-plugin-request-location-accuracy#12]
Hi again,

Previously I was just running this GPS check.  It begin loading the map, see the gps not enable, shows a blank map (because there are no gps coords yet), ask the user if they wanted gps enabled and then take them to their settings page where they manually enabled the GPS.  Upon doing so they would route back to the app and the map would continue its load based on new gps data.

```
function gpsEnable() {
  CheckGPS.check(function() {
    errMgmt("app/run",200,"GPS is enabled") ;
  },
  function() {
    errMgmt("app/run",201,"GPS is not enabled") ;
    var gpsMessage = "GPS is currently DISABLED.\n\nThis App requires the use of your devices GPS, you will next be directed to your devices settings page to turn on its GPS." ;
    cordova.dialogGPS(gpsMessage) ;
  }) ;
}
```

But now I have implemented the following, it starts to load the map page, see's gps is disabled, shows a blank map (because no gps data yet), asks user to enable GPS - when clicking yes the GPS simply auto-enables but then the map page never loads....just stays blank.  when the user manually enters in an address then the map loads.  

```

function gpsEnable() {
  CheckGPS.check(function() {
    errMgmt("app/run",200,"GPS is enabled") ;
  },
  function() {
    errMgmt("app/run",201,"GPS is not enabled") ;
    var gpsMessage = "GPS is currently DISABLED.\n\nThis App requires the use of your devices GPS, you will next be directed to your devices settings page to turn on its GPS." ;
    cordova.dialogGPS(gpsMessage) ;
  }) ;
}

function gpsSuccess(success){
  errMgmt("app/gpsEnable",1053,"successfully enabled") ;
}

function gpsFailure(error){
  errMgmt("app/gpsEnable",1054,"code: "+error.code + "; msg = "+error.message) ;
    if(error.code !== cordova.plugins.locationAccuracy.ERROR_USER_DISAGREED){
        gpsEnable() ;
    }
}
function globalGPS(){
    if(device.platform == "iOS"){
        gpsEnable() ;
    }else if(device.platform == "Android"){
        cordova.plugins.locationAccuracy.request(gpsSuccess, gpsFailure, cordova.plugins.locationAccuracy.REQUEST_PRIORITY_HIGH_ACCURACY);
    }
}
```

If I stop and restart the app (gps is already enabled now) then the map page loads just fine.

```

.run(function($ionicPlatform) {
  $ionicPlatform.ready(function() {
    // Hide the accessory bar by default (remove this to show the accessory bar above the keyboard
    // for form inputs)
    if (window.cordova && window.cordova.plugins && window.cordova.plugins.Keyboard) {
      cordova.plugins.Keyboard.hideKeyboardAccessoryBar(true);
    }
    if (window.StatusBar) {
      // org.apache.cordova.statusbar required
      StatusBar.styleLightContent();
    }   
    globalGPS() ;

  });

```

My map page auto-loads from the .state definitions, when app.js is loaded if a state isn't defined then default to:  `$urlRouterProvider.otherwise('/tab/map');`  which points to:

```
  .state('tab.map', {
    url: '/map',
    views: {
      'tab-map': {
        templateUrl: 'templates/tab-map.html',
        controller: 'MapCtrl'
      }
    }
  })
```

And in the MapCtrl:

```
.controller('MapCtrl', function($scope,$rootScope,constants) {
  ionic.Platform.ready(function() {
    setPickup("",0);
    $scope.map = setMap;    
  });
})
```

// setPickup is an external global function, its grabbing the local gpsCoords

**Comment by dpa99c** [comment]
Without knowing how you've implemented the map, I'm just speculating, but it's possible that whatever plugin/library you are using for the map hooks into the Cordova "resume" event to refresh the map. Hence, when you left the app to go to the settings page, switching back to the app would trigger the resume event. Now you're using this plugin to switch on GPS from within the app, there is no "resume" event being fired.

I would look into how you can manually reload/refresh the map once GPS has been switched on, depending on exactly how it is implemented.

**Comment by dpa99c** [comment]
FYI, I just published an update to this plugin (v2.0.0) which enables the requesting of an equivalent dialog (the best that can be achieved) on iOS

**Comment by rolinger** [comment]
First.  We can shut down this issue.  Not an issue at all, it was my code.  I was running two instances of $ionicPlatform.ready.   $ionicPlatform.ready in the app.js (.state definitions) and a second one ionic.Platform.ready in the map page controller.  The app one was firing first, checking for gps, but the 2nd one was also firing but was the one calling for google maps to be loaded....because gps wasn't enabled yet, google maps wasn't getting any coordinates and thus not properly loading (well, not loading at all). 

Essentially, it was a daisy chain of bugs all exposed by fixing how GPS is enabled to begin with. I pretty much have them all fixed now - I collapsed all GPS functions into the first app.js $ionicPlatform.ready and completely removed the ionic.Platform.ready from the map controller - which also eliminated a 2nd bug causing the splash screen to load twice.  

Second.  Very cool.  I will look into the v2.0.0 - I assume its documented now as well?  And I just completed an elaborate function that accounted for iOS GPS check.  However, I am certain yours is better.  lol
